src/v/features/feature_table.cc

```diff
@@ -297,6 +301,30 @@ class feature_table::probe {
     metrics::public_metric_groups _public_metrics;
 };
 
+namespace {
+
+security::license
+make_builtin_trial_license(security::license::clock::time_point start_time) {
+    auto expiry_time = start_time + std::chrono::days{45};
+    auto expiry = std::chrono::duration_cast<std::chrono::seconds>(
+      expiry_time.time_since_epoch());
+
+    if (std::getenv("__REDPANDA_DISABLE_BUILTIN_TRIAL_LICENSE") != nullptr) {
```

Comment:
the double prefix underscore is odd. we have so many notations we should unify. 

RPK has env variables for example as well that do not have this. @twmb - we should have a developer experience here unify these. the product flags feel disjointed.

Reply:
The intended use for this environmental variable is for testing and not for end-user consumption.  This pattern matches other environmental variables [here](https://github.com/redpanda-data/redpanda/blob/64375e957eecf4eff7ae4f63633c868046037854/src/v/features/feature_table.cc#L334), [here](https://github.com/redpanda-data/redpanda/blob/64375e957eecf4eff7ae4f63633c868046037854/src/v/features/feature_table.cc#L352)], and [here](https://github.com/redpanda-data/redpanda/blob/64375e957eecf4eff7ae4f63633c868046037854/src/v/features/feature_table.cc#L394)

Reply:
i pinged @mattschumpert on this too to make sure all the vars are consistent. this should have product input for consistency. i only care about consistency, the prefix,postfix,dot-notation,etc doesn't matter as much to me as consistency across all product lines.

Reply:
The ENV vars specifically are so internal facing and no user ever sees them unless their browsing the code that it only matters for redpanda devs I think. 

I've never once looked at any of these (only a couple of documented external ones).

the '__REDPANDA' seems to be our convention for internal, and seems consistent so it makes sense to me.

Reply:
Yeh, internal consumption only. Should never be in documentation.